ca/certificate-authority.go

```diff
@@ -309,7 +380,11 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 	}
 
 	certPEM, err := ca.Signer.Sign(req)
+	ca.noteHSMFault(err)
 	if err != nil {
+		// Since we just used the HSM (if we're using one), note whether it worked
```

Comment:
I think this comment is in the wrong place? Also the function only notes whether the Sign request failed. I think that's the right behavior, but the comment needs update.